server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1861,8 +1861,10 @@ private CompactionStats _majorCompact(MajorCompactionReason reason) throws IOExc
       if (inputFiles.isEmpty()) {
         if (reason == MajorCompactionReason.USER) {
           // no work to do
-          lastCompactID = compactionId.getFirst();
-          updateCompactionID = true;
+//          lastCompactID = compactionId.getFirst();
```

Comment:
I'd definitely feel more comfortable if @keith-turner reviewed it. He's the last person I know who fixed a bug in this code.